server/cluster/cluster.go

```diff
@@ -1594,24 +1601,42 @@ func getStoreTopoWeight(store *core.StoreInfo, stores []*core.StoreInfo, locatio
 	return weight / sameLocationStoreNum
 }
 
-func buildTopology(s *core.StoreInfo, stores []*core.StoreInfo, locationLabels []string) (map[string]interface{}, float64) {
+func buildTopology(s *core.StoreInfo, stores []*core.StoreInfo, locationLabels []string, count int) (map[string]interface{}, []string, float64, bool) {
 	topology := make(map[string]interface{})
 	sameLocationStoreNum := 1.0
+	totalLabelCount := make([]int, len(locationLabels))
 	for _, store := range stores {
 		if store.IsServing() || store.IsPreparing() {
-			updateTopology(topology, getSortedLabels(store.GetLabels(), locationLabels))
+			labelCount := updateTopology(topology, getSortedLabels(store.GetLabels(), locationLabels))
+			for i, c := range labelCount {
+				totalLabelCount[i] += c
+			}
 		}
+	}
 
+	validLabels := locationLabels
+	var isMatch bool
+	for i, c := range totalLabelCount {
+		if count/c == 0 {
+			validLabels = validLabels[:i]
+			break
+		}
+		if count/c == 1 {
+			validLabels = []string{validLabels[i]}
```

Comment:
a store with label(dc1,zone1) and a store with label(dc2 zone1) are different. So I think it should be 
```
validLabels = validLabels[:i+1]
```
Otherwise, it will affect the results of CompareLocation.